File: familyPlanner/src/models/kassalappModel.ts

export interface KassalappProduct {
    id: number;
    name: string;
    brand: string;
    vendor: string;
    ean?: string;
    url?: string;
    image?: string;
    description?: string;
    ingredients?: string;
    currentPrice: number;
    currentUnitPrice?: number;
    weight?: number;
    weightUnit?: string;
    store?: Store;
    priceHistory?: PriceHistory[];
    allergens?: Allergens[];
    nutrition?: Nutrition[];
    createdAt?: Date;
    updatedAt?: Date;
}

interface PriceHistory {
    price?: number;
    date?: Date;
}
interface Store {
    name?: string;
    code?: string;
    url?: string;
    logo?: string;
}

interface Allergens {
    code?: string;
    displayName?: string;
    contains?: string;
}

interface Nutrition {
    code?: string;
    displayName?: string;
    amount?: number;
    unit?: string;
}

File: backend/models/KassalappModel/KassalappModel.cs

namespace FamilyPlanner.Models.KassalappModel {
    public sealed class KassalappModel {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Vendor { get; set; }
        public string? EAN { get; set; }
        public string? Url { get; set; }
        public string? ImageUri { get; set; }
        public string? Description { get; set; }
        public string? Ingredients { get; set; }
        public decimal CurrentPrice { get; set; }
        public decimal CurrentUnitPrice { get; set; }
        public decimal? Weight { get; set; }
        public KassalappStore? Store { get; set; }
        public List<KassalappPriceHistory>? PriceHistory { get; set; }
        public List<KassalappAllergen>? Allergens { get; set; }
        public List<KassalappNutrition>? Nutritions { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}